#!/usr/bin/env python
PACKAGE = "task_switch"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("int_param", int_t, 0, "An Integer parameter", 50, 0, 100)
gen.add("maxEnergy", int_t, 0, "A Integer parameter", 4000, 0, 5000)
gen.add("minEnergy", int_t, 0, "A Integer parameter", 1500, 0, 5000)
gen.add("Kd", int_t, 0, "A Integer parameter", 50, 0, 200)
gen.add("k_charge", double_t, 0, "A double parameter", 0.15, 0.01, 1.0)
# gen.add("delta_decrease", double_t, 0, "A double parameter", 1.0, 0, 5)
# gen.add("delta_increase", double_t, 0, "A double parameter", 0.01, 0, 0.1)
# gen.add("str_param", str_t, 0, "A string parameter", "Hello World")
# gen.add("bool_param", bool_t, 0, "A Boolean parameter", True)


exit(gen.generate(PACKAGE, "task_switch", "charge_params"))
